plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java

```diff
@@ -778,6 +790,35 @@ public Object[][] unsupportedTimestampDataProvider()
         };
     }
 
+    @Test(dataProvider = "sessionZonesDataProvider")
+    public void testClickHouseDateTimeWithTimeZone(ZoneId sessionZone)
+    {
+        Session session = Session.builder(getSession())
+                .setTimeZoneKey(TimeZoneKey.getTimeZoneKey(sessionZone.getId()))
+                .build();
+
+        dateTimeWithTimeZoneTest(clickhouseDateTimeInputTypeFactory("datetime"), clickhouseDateTimeInputLiteralFactory(), trinoTimestampExpectedLiteralFactory())
+                .execute(getQueryRunner(), session, clickhouseCreateAndInsert("tpch.datetime_tz"));
+    }
+
+    private SqlDataTypeTest dateTimeWithTimeZoneTest(Function<ZoneId, String> inputTypeFactory, BiFunction<LocalDateTime, ZoneId, String> inputLiteralFactory, BiFunction<LocalDateTime, ZoneId, String> expectedLiteralFactory)
+    {
+        ZoneId utc = ZoneId.of("UTC");
+        SqlDataTypeTest tests = SqlDataTypeTest.create()
+                .addRoundTrip(format("Nullable(%s)", inputTypeFactory.apply(utc)), "NULL", TIMESTAMP_TZ_SECONDS, "CAST(NULL AS TIMESTAMP(0) WITH TIME ZONE)");
+
+        for (ZoneId zoneId : ImmutableList.of(utc, kathmandu)) {
```

Comment:
since you're using kathmandu zone it might be also useful to actually test with a timestamp which has gap in that zone.

Reply:
will do.